### PR TITLE
Switch from SQLTypeInfo to hdk::ir::Type (parts 21-23)

### PR DESCRIPTION
**Pull request by ienkovich**

### CODE REVIEW

File: QueryEngine/ConstantIR.cpp [hunk 1]
```diff
@@ -18,79 +18,14 @@
...
+                                                 bool use_dict_encoding,
+                                                 int dict_id,
                                                  const CompilationOptions& co) {
   AUTOMATIC_IR_METADATA(cgen_state_);
   if (co.hoist_literals) {
     std::vector<const hdk::ir::Constant*> constants(
         executor()->deviceCount(co.device_type), constant);
-    return codegenHoistedConstants(constants, enc_type, dict_id);
-  }
-  const auto& type_info = constant->get_type_info();
```
Comment (alexbaden) [inline QueryEngine/ConstantIR.cpp:30]: I am trying to follow where the non-hoisted listerals side of the if went. It looks like there was some merging with another function, but I don't see the function signature change in the header (I am looking at this in github diff mode and not in the IDE). Have you looked at the non-hoisted literals side in testing these branches?

Comment (ienkovich) [reply QueryEngine/ConstantIR.cpp:30]: Here methods `codegen(const hdk::ir::Constant*, const hdk::ir::Type*, const CompilationOptions&);` and `codegen(const hdk::ir::Constant*, EncodingType, int, const CompilationOptions&)` are merged into a single one which takes `bool` and `int` for dictionary encoding. Signature changes are in `CodeGenerator.h:126-129`

Comment (alexbaden) [reply QueryEngine/ConstantIR.cpp:30]: Thanks,  with the rebase it is clear now.

**Review by alexbaden — APPROVED** [review]

**Comment by ienkovich** [comment]
merged

**Pull request closed without merging**